### PR TITLE
Store user_webdavauth users in DB + code cleanup

### PR DESCRIPTION
**Pull request by LukasReschke**
This commit introduces the storing of user_webdavauth users in the database which is a pre-requisite of https://github.com/owncloud/core/issues/12620. 

Furthermore, I refactored the code and removed deprecated code, as a little gimmick I added unit tests for the backend. - Especially considering that some people are using this as reference implementation for an authentication backend I consider it important that this backend follows best-practises. (still not perfect - but way better)

@MorrisJobke Please review.

@karlitschek Thought you might like that since you created that app. Care to review as well? - Don't worry, most of the change are unit tests :-)

### CODE REVIEW

**Comment by MorrisJobke** [comment]
@owncloud-bot retest this please

**Comment by MorrisJobke** [comment]
@LukasReschke Any news here?

**Comment by LukasReschke** [comment]
I think it's better to keep this for 8.1 now.

File: apps/user_webdavauth/appinfo/database.xml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+			<field>
+				<name>displayname</name>
+				<type>text</type>
+				<notnull>false</notnull>
+				<length>255</length>
+			</field>
+
+			<index>
+				<name>webdav_uid</name>
```
Comment (LukasReschke) [inline apps/user_webdavauth/appinfo/database.xml]: Reminder to myself: Learn more about that mysterious database.xml - THX! ;-)

**Comment by LukasReschke** [comment]
Rebased. Ready to review.

File: apps/user_webdavauth/appinfo/database.xml (hunk 1 above)
Comment (DeepDiver1975) [inline apps/user_webdavauth/appinfo/database.xml]: > Done

???

File: lib/private/server.php [hunk 2]
```diff
@@ -708,7 +708,9 @@ function getWebRoot() {
 	}
 
 	/**
-	 * @return \OCP\IDateTimeZone
+	 * Get the timezone of the current user, based on his session information and config data
+	 *
+	 * @return \DateTimeZone
```
Comment (DeepDiver1975) [inline lib/private/server.php]: the server is meant to return ocp interfaces

Comment (LukasReschke) [inline lib/private/server.php]: :see_no_evil:

File: apps/user_webdavauth/appinfo/database.xml (hunk 1 above)
Comment (MorrisJobke) [inline apps/user_webdavauth/appinfo/database.xml]: I thought I added a commit to this branch o.O

Comment (LukasReschke) [inline apps/user_webdavauth/appinfo/database.xml]: Database noob here: Isn't the `unique` one line below exactly the same? :see_no_evil:

**Comment by LukasReschke** [comment]
Rebased as well.

File: apps/user_webdavauth/appinfo/database.xml (hunk 1 above)
Comment (MorrisJobke) [inline apps/user_webdavauth/appinfo/database.xml]: Unique != primary ;)

**Comment by canepan** [comment]
Hello.
I'm wondering if it is possible to make this patch (and external users management) more useful.
Right now, an user which logs in via webdav does not belong to any group, has no email or display name, so apart from storing files and sharing via link or directly with other users.
It's not possible to assign quota, or do anything with external users.

I think this is the same for all external authentication methods (apart from LDAP, which can map LDAP groups to OC groups, I guess)

**Comment by MorrisJobke** [comment]
@DeepDiver1975 @LukasReschke Still something for 8.1?

**Comment by MorrisJobke** [comment]
@LukasReschke You said, that you want to have a look at this :)

**Comment by DeepDiver1975** [comment]
@LukasReschke 9.0 ????

**Comment by LukasReschke** [comment]
Replaced by https://github.com/owncloud/core/pull/19229 then I guess.

**Comment by DeepDiver1975** [comment]
> Replaced by #19229 then I guess.

yep

**Pull request closed without merging**